src/librustc_mir/interpret/cast.rs

```diff
@@ -85,7 +85,7 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                             self.param_env,
                             def_id,
                             substs,
-                        ).ok_or_else(|| InterpError::TooGeneric.into());
+                        ).ok_or_else(|| err_inval!(TooGeneric).into());
```

Comment:
I think this `into ` can be removed

Reply:
```rust
error[E0308]: mismatched types
  --> src/librustc_mir/interpret/cast.rs:83:63
   |
83 |                           let instance: InterpResult<'tcx, _> = ty::Instance::resolve(
   |  _______________________________________________________________^
84 | |                             *self.tcx,
85 | |                             self.param_env,
86 | |                             def_id,
87 | |                             substs,
88 | |                         ).ok_or_else(|| err_inval!(TooGeneric));
   | |_______________________________________________________________^ expected struct `rustc::mir::interpret::InterpErrorInfo`, found enum `rustc::mir::interpret::InterpError`
   |
   = note: expected type `std::result::Result<_, rustc::mir::interpret::InterpErrorInfo<'tcx>>`
              found type `std::result::Result<rustc::ty::Instance<'_>, rustc::mir::interpret::InterpError<'_>>`
```

Looks like its needed?

Reply:
This is funny, the `?` is delayed until after the binder.

`.ok_or_else(|| err_inval!(TooGeneric))?` should work (and removing the type ascription from `instance`).

Reply:
Done.